_blog/2024-01-18-parsing-json-rust.mdx

```diff
@@ -0,0 +1,229 @@
+---
+title: "Parsing JSON in Rust"
+description: This article talks about parsing JSON in Rust and compares JSON parsing libraries. 
+thumb: bypass-aws-complexity-thumb.png 
+cover: bypass-aws-complexity-thumb.png 
+date: '2024-01-11T14:30:00'
+---
+In this article we’re going to talk about how to use JSON parsing libraries in Rust, as well as a comparison of the most popular libraries and how they perform.
+
+## Using JSON Parsing Libraries in Rust
+
+### Generating JSON manually in Rust
+
+To get started with working with JSON in Rust, you’ll want to install a library that lets you manipulate JSON easily. One of the popular crates currently available to use is `serde-json`. You can install it by running the following:
+
+```rust
+cargo add serde-json
+```
+
+Once done, you can create JSON manually like this:
+
+```rust
+use serde_json::{Result, Value};
+
+fn untyped_example() -> Result<()> {
+    // Some JSON input data as a &str. Maybe this comes from the user.
+    let data = r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "phones": [
+                "+44 1234567",
+                "+44 2345678"
+            ]
+        }"#;
+
+    // Parse the string of data into serde_json::Value.
+    let v: Value = serde_json::from_str(data)?;
+
+    // Access parts of the data by indexing with square brackets.
+    println!("Please call {} at the number {}", v["name"], v["phones"][0]);
+
+    Ok(())
+}
+```
+
+However, what you *probably* want to do is be able to serialize data to a struct (creating a struct from JSON), as well as deserialize data from a struct (creating JSON from a struct). Let’s have a look at this in the next section.
```

Comment:
I think when I was writing this I meant in terms of storing the JSON data as a struct instead of raw data. I'll amend this so it actually makes sense for the article